core/services/gateway/handlers/functions/orm.go

```diff
@@ -130,3 +135,40 @@ func (cs *cachedSubscriptionRow) encode() CachedSubscription {
 		},
 	}
 }
+
+func (o *orm) GetAllowedSenders(offset, limit uint, qopts ...pg.QOpt) ([]common.Address, error) {
+	var addresses []common.Address
+	stmt := fmt.Sprintf(`
+		SELECT allowed_address
+		FROM %s
+		WHERE router_contract_address = $1
+		ORDER BY id ASC
+		OFFSET $2
+		LIMIT $3;
+	`, allowlistTableName)
+	err := o.q.WithOpts(qopts...).Select(&addresses, stmt, o.routerContractAddress, offset, limit)
+	if err != nil {
+		return addresses, err
+	}
+	o.lggr.Debugf("Successfully fetched allowed sender list from DB. offset: %d, limit: %d, length: %d", offset, limit, len(addresses))
+
+	return addresses, nil
+}
+
+func (o *orm) CreateAllowedSender(allowedSender common.Address, qopts ...pg.QOpt) error {
```

Comment:
Would it make sense to do this with batches of allowed senders instead of one-at-a-time?

Reply:
🤔 I think you are right, before we needed to iterate to generate the index. now we are taking care of this at the db level by using a auto incremental value.